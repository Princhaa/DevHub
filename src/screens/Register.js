import React, { Component } from 'react'
import { View, Text, StyleSheet, ActivityIndicator } from 'react-native'

import CustomButton from '../components/CustomButton'
import CustomTextInput from '../components/CustomTextInput'
import constants from '../config/constants';

export default class Register extends Component {
  static navigationOptions = {
    header: null
  }

  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: '',
      confirmPassword: '',
      isRegistering: false
    }

    this.emailRef = React.createRef()
    this.passwordRef = React.createRef()
    this.confirmPasswordRef = React.createRef()

    this._handleTextInputChange = this._handleTextInputChange.bind(this)
    this._handleSubmitEditing = this._handleSubmitEditing.bind(this)
    this._handleRegisterPress = this._handleRegisterPress.bind(this)
  }

  _handleTextInputChange(state, value) {
    this.setState({ [state]: value })
  }

  _handleSubmitEditing(nextRef) {
    if (typeof nextRef === 'string') {
      this._handleRegisterPress()
      return
    }
    nextRef.current.focus()
  }

  _handleRegisterPress() {
    this.setState({ isRegistering: !this.state.isRegistering })
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}> DevHub </Text>
        <CustomTextInput
          ref={this.emailRef}
          placeholder={'Email'}
          keyboardType={'email-address'}
          autoCapitalize={'none'}
          returnKeyType={'next'}
          onChangeText={this._handleTextInputChange.bind(this, 'email')}
          onSubmitEditing={this._handleSubmitEditing.bind(this, this.passwordRef)}
        />
        <CustomTextInput
          ref={this.passwordRef}
          placeholder={'Password'}
          autoCapitalize={'none'}
          returnKeyType={'next'}
          onChangeText={this._handleTextInputChange.bind(this, 'password')}
          onSubmitEditing={this._handleSubmitEditing.bind(this, this.confirmPasswordRef)}
        />
        <CustomTextInput
          ref={this.confirmPasswordRef}
          placeholder={'Confirm Password'}
          autoCapitalize={'none'}
          returnKeyType={'next'}
          onChangeText={this._handleTextInputChange.bind(this, 'confirmPassword')}
          onSubmitEditing={this._handleSubmitEditing.bind(this, 'done')}
        />
        <CustomButton color={constants.PRIMARY_COLOR} onPress={this._handleRegisterPress}>
          {this.state.isRegistering ? (
            <ActivityIndicator color={'white'}/>
          ) : (
            <Text style={{ color: 'white' }}>Register</Text>
          )}
        </CustomButton>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    padding: 20,
    backgroundColor: 'white'
  },

  title: {
    fontSize: 30,
    fontWeight: 'bold',
    marginVertical: 10
  }
})
