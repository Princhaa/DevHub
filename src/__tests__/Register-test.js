import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Register from '../screens/Register'
import { shallow } from 'enzyme'

it('Renders Register Correctly',  () => {
  const renderer = new ShallowRenderer()
  renderer.render(<Register />)
  const tree = renderer.getRenderOutput()
  expect(tree).toMatchSnapshot()
})

describe('Testing Register button to change state', () => {
  it('State changes as expected', () => {
    const wrapper = shallow(<Register />)
    expect(wrapper).toMatchSnapshot()
    wrapper.instance()._handleRegisterPress()
    expect(wrapper.state().isRegistering).toEqual(true)
  })
})

describe('Changing text input state', () => {
  it('State changes as expected', () => {
    const wrapper = shallow(<Register />)
    expect(wrapper).toMatchSnapshot()
    wrapper.instance()._handleTextInputChange('email', 'bob@mail.com')
    expect(wrapper.state().email).toEqual('bob@mail.com')
  })
})

describe('Changing isRegistering state', () => {
  it('State changes as expected', () => {
    const mockRef = {
      current: {
        focus: jest.fn()
      }
    }
    const wrapper = shallow(<Register />)
    expect(wrapper).toMatchSnapshot()
    wrapper.instance()._handleSubmitEditing('done')
    expect(wrapper.state().isRegistering).toEqual(true)
    wrapper.instance()._handleSubmitEditing(mockRef)
    expect(mockRef.current.focus).toBeCalled()
  })
})