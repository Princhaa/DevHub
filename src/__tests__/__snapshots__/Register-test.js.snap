// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Changing isRegistering state State changes as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 30,
              "fontWeight": "bold",
              "marginVertical": 10,
            }
          }
        >
           DevHub 
        </Text>,
        <CustomTextInput
          autoCapitalize="none"
          keyboardType="email-address"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Email"
          returnKeyType="next"
        />,
        <CustomTextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Password"
          returnKeyType="next"
        />,
        <CustomTextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Confirm Password"
          returnKeyType="next"
        />,
        <CustomButton
          color="#39A9DB"
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
              }
            }
          >
            Register
          </Text>
        </CustomButton>,
      ],
      "style": Object {
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": " DevHub ",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "marginVertical": 10,
          },
        },
        "ref": null,
        "rendered": " DevHub ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoCapitalize": "none",
          "keyboardType": "email-address",
          "onChangeText": [Function],
          "onSubmitEditing": [Function],
          "placeholder": "Email",
          "returnKeyType": "next",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoCapitalize": "none",
          "onChangeText": [Function],
          "onSubmitEditing": [Function],
          "placeholder": "Password",
          "returnKeyType": "next",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoCapitalize": "none",
          "onChangeText": [Function],
          "onSubmitEditing": [Function],
          "placeholder": "Confirm Password",
          "returnKeyType": "next",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
              }
            }
          >
            Register
          </Text>,
          "color": "#39A9DB",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Register",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "white",
            },
          },
          "ref": null,
          "rendered": "Register",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
                "marginVertical": 10,
              }
            }
          >
             DevHub 
          </Text>,
          <CustomTextInput
            autoCapitalize="none"
            keyboardType="email-address"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Email"
            returnKeyType="next"
          />,
          <CustomTextInput
            autoCapitalize="none"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Password"
            returnKeyType="next"
          />,
          <CustomTextInput
            autoCapitalize="none"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Confirm Password"
            returnKeyType="next"
          />,
          <CustomButton
            color="#39A9DB"
            onPress={[Function]}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Register
            </Text>
          </CustomButton>,
        ],
        "style": Object {
          "backgroundColor": "white",
          "flex": 1,
          "justifyContent": "center",
          "padding": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": " DevHub ",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 30,
              "fontWeight": "bold",
              "marginVertical": 10,
            },
          },
          "ref": null,
          "rendered": " DevHub ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoCapitalize": "none",
            "keyboardType": "email-address",
            "onChangeText": [Function],
            "onSubmitEditing": [Function],
            "placeholder": "Email",
            "returnKeyType": "next",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoCapitalize": "none",
            "onChangeText": [Function],
            "onSubmitEditing": [Function],
            "placeholder": "Password",
            "returnKeyType": "next",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoCapitalize": "none",
            "onChangeText": [Function],
            "onSubmitEditing": [Function],
            "placeholder": "Confirm Password",
            "returnKeyType": "next",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Register
            </Text>,
            "color": "#39A9DB",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Register",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "white",
              },
            },
            "ref": null,
            "rendered": "Register",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Changing text input state State changes as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 30,
              "fontWeight": "bold",
              "marginVertical": 10,
            }
          }
        >
           DevHub 
        </Text>,
        <CustomTextInput
          autoCapitalize="none"
          keyboardType="email-address"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Email"
          returnKeyType="next"
        />,
        <CustomTextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Password"
          returnKeyType="next"
        />,
        <CustomTextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Confirm Password"
          returnKeyType="next"
        />,
        <CustomButton
          color="#39A9DB"
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
              }
            }
          >
            Register
          </Text>
        </CustomButton>,
      ],
      "style": Object {
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": " DevHub ",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "marginVertical": 10,
          },
        },
        "ref": null,
        "rendered": " DevHub ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoCapitalize": "none",
          "keyboardType": "email-address",
          "onChangeText": [Function],
          "onSubmitEditing": [Function],
          "placeholder": "Email",
          "returnKeyType": "next",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoCapitalize": "none",
          "onChangeText": [Function],
          "onSubmitEditing": [Function],
          "placeholder": "Password",
          "returnKeyType": "next",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoCapitalize": "none",
          "onChangeText": [Function],
          "onSubmitEditing": [Function],
          "placeholder": "Confirm Password",
          "returnKeyType": "next",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
              }
            }
          >
            Register
          </Text>,
          "color": "#39A9DB",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Register",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "white",
            },
          },
          "ref": null,
          "rendered": "Register",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
                "marginVertical": 10,
              }
            }
          >
             DevHub 
          </Text>,
          <CustomTextInput
            autoCapitalize="none"
            keyboardType="email-address"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Email"
            returnKeyType="next"
          />,
          <CustomTextInput
            autoCapitalize="none"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Password"
            returnKeyType="next"
          />,
          <CustomTextInput
            autoCapitalize="none"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Confirm Password"
            returnKeyType="next"
          />,
          <CustomButton
            color="#39A9DB"
            onPress={[Function]}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Register
            </Text>
          </CustomButton>,
        ],
        "style": Object {
          "backgroundColor": "white",
          "flex": 1,
          "justifyContent": "center",
          "padding": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": " DevHub ",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 30,
              "fontWeight": "bold",
              "marginVertical": 10,
            },
          },
          "ref": null,
          "rendered": " DevHub ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoCapitalize": "none",
            "keyboardType": "email-address",
            "onChangeText": [Function],
            "onSubmitEditing": [Function],
            "placeholder": "Email",
            "returnKeyType": "next",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoCapitalize": "none",
            "onChangeText": [Function],
            "onSubmitEditing": [Function],
            "placeholder": "Password",
            "returnKeyType": "next",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoCapitalize": "none",
            "onChangeText": [Function],
            "onSubmitEditing": [Function],
            "placeholder": "Confirm Password",
            "returnKeyType": "next",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Register
            </Text>,
            "color": "#39A9DB",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Register",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "white",
              },
            },
            "ref": null,
            "rendered": "Register",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Renders Register Correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 30,
        "fontWeight": "bold",
        "marginVertical": 10,
      }
    }
  >
     DevHub 
  </Text>
  <CustomTextInput
    autoCapitalize="none"
    keyboardType="email-address"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Email"
    returnKeyType="next"
  />
  <CustomTextInput
    autoCapitalize="none"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Password"
    returnKeyType="next"
  />
  <CustomTextInput
    autoCapitalize="none"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Confirm Password"
    returnKeyType="next"
  />
  <CustomButton
    color="#39A9DB"
    onPress={[Function]}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      Register
    </Text>
  </CustomButton>
</Component>
`;

exports[`Testing Register button to change state State changes as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 30,
              "fontWeight": "bold",
              "marginVertical": 10,
            }
          }
        >
           DevHub 
        </Text>,
        <CustomTextInput
          autoCapitalize="none"
          keyboardType="email-address"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Email"
          returnKeyType="next"
        />,
        <CustomTextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Password"
          returnKeyType="next"
        />,
        <CustomTextInput
          autoCapitalize="none"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Confirm Password"
          returnKeyType="next"
        />,
        <CustomButton
          color="#39A9DB"
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
              }
            }
          >
            Register
          </Text>
        </CustomButton>,
      ],
      "style": Object {
        "backgroundColor": "white",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": " DevHub ",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "marginVertical": 10,
          },
        },
        "ref": null,
        "rendered": " DevHub ",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoCapitalize": "none",
          "keyboardType": "email-address",
          "onChangeText": [Function],
          "onSubmitEditing": [Function],
          "placeholder": "Email",
          "returnKeyType": "next",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoCapitalize": "none",
          "onChangeText": [Function],
          "onSubmitEditing": [Function],
          "placeholder": "Password",
          "returnKeyType": "next",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoCapitalize": "none",
          "onChangeText": [Function],
          "onSubmitEditing": [Function],
          "placeholder": "Confirm Password",
          "returnKeyType": "next",
        },
        "ref": Object {
          "current": null,
        },
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
              }
            }
          >
            Register
          </Text>,
          "color": "#39A9DB",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Register",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "white",
            },
          },
          "ref": null,
          "rendered": "Register",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "bold",
                "marginVertical": 10,
              }
            }
          >
             DevHub 
          </Text>,
          <CustomTextInput
            autoCapitalize="none"
            keyboardType="email-address"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Email"
            returnKeyType="next"
          />,
          <CustomTextInput
            autoCapitalize="none"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Password"
            returnKeyType="next"
          />,
          <CustomTextInput
            autoCapitalize="none"
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Confirm Password"
            returnKeyType="next"
          />,
          <CustomButton
            color="#39A9DB"
            onPress={[Function]}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Register
            </Text>
          </CustomButton>,
        ],
        "style": Object {
          "backgroundColor": "white",
          "flex": 1,
          "justifyContent": "center",
          "padding": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": " DevHub ",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 30,
              "fontWeight": "bold",
              "marginVertical": 10,
            },
          },
          "ref": null,
          "rendered": " DevHub ",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoCapitalize": "none",
            "keyboardType": "email-address",
            "onChangeText": [Function],
            "onSubmitEditing": [Function],
            "placeholder": "Email",
            "returnKeyType": "next",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoCapitalize": "none",
            "onChangeText": [Function],
            "onSubmitEditing": [Function],
            "placeholder": "Password",
            "returnKeyType": "next",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoCapitalize": "none",
            "onChangeText": [Function],
            "onSubmitEditing": [Function],
            "placeholder": "Confirm Password",
            "returnKeyType": "next",
          },
          "ref": Object {
            "current": null,
          },
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Register
            </Text>,
            "color": "#39A9DB",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Register",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "white",
              },
            },
            "ref": null,
            "rendered": "Register",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
